src/libs/OptionsListUtils.js

```diff
@@ -847,31 +847,34 @@ function getCategoryListSections(categories, recentlyUsedCategories, selectedOpt
         return categorySections;
     }
 
-    if (numberOfCategories < CONST.CATEGORY_LIST_THRESHOLD) {
+    if (!_.isEmpty(selectedOptions)) {
         categorySections.push({
-            // "All" section when items amount less than the threshold
+            // "Selected" section
             title: '',
             shouldShow: false,
             indexOffset,
-            data: getCategoryOptionTree(enabledCategories),
+            data: getCategoryOptionTree(selectedOptions, true),
         });
 
-        return categorySections;
+        indexOffset += selectedOptions.length;
     }
 
-    if (!_.isEmpty(selectedOptions)) {
+    const selectedOptionNames = _.map(selectedOptions, (selectedOption) => selectedOption.name);
+    const filteredCategories = _.filter(enabledCategories, (category) => !_.includes(selectedOptionNames, category.name));
+    const numberOfVisibleCategories = selectedOptions.length + filteredCategories.length;
```

Comment:
We should have checked if the `selectedOptions` were enabled or not. This later caused https://github.com/Expensify/App/issues/37766